src/state/slices/opportunitiesSlice/opportunitiesSlice.ts

```diff
@@ -0,0 +1,117 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { createApi } from '@reduxjs/toolkit/query/react'
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import type { DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { BASE_RTK_CREATE_API_CONFIG } from 'state/apis/const'
+
+export const initialState: OpportunitiesState = {
+  lp: {
+    byAccountId: {},
+    byId: {},
+    ids: [],
+  },
+  farming: {
+    byAccountId: {},
+    byId: {},
+    ids: [],
+  },
+  userLp: {
+    byId: {},
+    ids: [],
+  },
+  userFarming: {
+    byId: {},
+    ids: [],
+  },
+}
+
+// The contract info, also referred to as "metadata" - doesn't contain any user-specific data
```

Comment:
we should definitely establish this terminology now

* BareOpportunity
* BaseOpportunity
* OpportunityMetadata

i don't have a strong preference for any just think we should get this "bottom" type correct

Reply:
I have a strong preference for the last one, way less cognitive overhead as the verbiage clearly separates the opportunity metadata from the rest. You still get to have base / augmented types, but the mental model makes more sense (at least for me)

Reply:
or `AbstractOpportunity`... but it's already kinda concrete of itself

Reply:
fuck that idea ^^^